### DISCUSSION TITLE
Looping through a frame

### DISCUSSION THREAD

**Discussion by ToneseekerMusical** [TomSchimansky/CustomTkinter#1217] [Q&A] [answered]
Is it possible to loop through a frame to get the value of the inputs within the frame? I know I can get them individually, but I have over 100 fields to get, and I'd rather not hard code each item.

I've tried looping through `frame.children` which only returns an unusable string of the field type as it's just a descriptor of the field and not the actual field name. I've also tried looping through the frame itself, but that throws a typing error in `cget()` as it tries to concatenate an integer to a string. I've also tried looping through `frame.keys()` but that only returns the available properties that can be applied to a frame.

Any help would be greatly appreciated.

**Comment by Wolf-SO** [comment, marked as answer]
The widget property [`children` is a `dict`](https://docs.python.org/3/library/tkinter.html#tkinter.Tk.children), so you should work on its *values*:

    for key, widget in frame.children.items():
        print("key", key)
        print("widget", widget)

If your frame contains only `CTkEntry` widgets, you can print all current inputs like this:

    for key, widget in frame.children.items():
        print("key", key)
        print("value:", widget.get())

But if you loop over all `CTkEntry` widgets, how do you know what you get?

In my opinion, it's better to use a list and add each input you create to it. You can also do this by making a function that creates `CTkEntry` widgets for you. See following example:

```python
import customtkinter

app = customtkinter.CTk()

inputs = []

def add_input(parent, placeholder):
    input = customtkinter.CTkEntry(master=parent, placeholder_text=placeholder)
    input.pack(padx=8, pady=8)
    inputs.append(input)

for i in range(5):
    add_input(app, f"input {i+1}")

def print_inputs():
    for input in inputs:
        print("value:", input.get())

customtkinter.CTkButton(master=app, text="Print", command=print_inputs).pack(pady=8)

app.mainloop()
```

**Reply by ToneseekerMusical** [reply]
You are an absolute life saver! I'm interacting with a mongoDB so I need the fields to be named so they get inserted into the document properly,, but I should be able to just set inputs as a dict and loop through the key value pairs to prepare the insertion object. Thank you!
